### PR TITLE
Add suse distros support

### LINKED ISSUE #1416: Add support for OpenSuSE
If you run Briefcase on OpenSuSE, you currently get an error:

### Describe the bug

Running a built app fails with:
```
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
Python runtime state: core initialized
Install path: <a project specific path>
Pre-initializing Python runtime...
PYTHONPATH:
- /usr/lib/python3.11
- /usr/lib/python3.11/lib-dynload
- <some project specific paths>
Configure argc/argv...
Initializing Python runtime...
ModuleNotFoundError: No module named 'encodings'
```

And we have:
```
> python3.11 -c 'import encodings; print(encodings.__file__)'
/usr/lib64/python3.11/encodings/__init__.py
```

Note the path `/usr/lib64/python3.11` is not in PYTHONPATH (`lib64` instead of `lib`).

### Steps to reproduce

1. Get the basic template
2. Run `briefcase run`

### Expected behavior

Building picks up the paths required for python to run.

### Screenshots

_No response_

### Environment

- Operating System: OpenSUSE
- Python version: 3.11
- Software versions:
  - Briefcase: 0.3.15
  - Toga: 0.3.1

### Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Cimbali**
Add support for suse family of distros. Fixes #1416

The rolling release line in these distros, openSUSE Tumbleweed, has a valid `VERSION_ID` (which is the release date) so I have not done any special handling to return `"rolling"`. Here’s my /etc/os-release for reference:

```
NAME="openSUSE Tumbleweed"
# VERSION="20230818"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20230818"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20230818"
BUG_REPORT_URL="https://bugzilla.opensuse.org"
SUPPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"
```

## PR Checklist:

- [ ] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

Leaving boxes unticked as I’m not very familiar with the project so testing might be light, and not sure what documentation is appropriate (I’ve found not list of supported Linux distros to extend…)

### CODE REVIEW

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
I'm not sure if you're waiting for a code review on this - but it's currently failing some of the existing tests. 

Related to that test - given that `openSUSE` returns `opensuse-tumbleweed` as the vendor, we probably want to transform that into `tumbleweed` so that the TOML parsing doesn't choke on a configuration section named `[tool.briefcase.app.myapp.linux.system.suse.opensuse-tumbleweed]`.

In terms of documentation, [this](https://github.com/beeware/briefcase/blob/main/docs/reference/platforms/linux/system.rst) is the file that needs to updated.

The release note (1416.misc.rst) should also be named `1416.feature.rst`, as this is a new feature.

You'll also find that you'll at need some updates to the [briefcase-linux-system-template](https://github.com/beeware/briefcase-linux-system-template); as well as an update to [briefcase-template](https://github.com/beeware/briefcase-template) if you want openSUSE to work out of the box on new projects.

These are all relative minor changes, and I would have been able to make them on your behalf - but you've submitted the PR from your `main` branch rather than a feature branch, and as a result, it's not editable by maintainers.

**Comment by Cimbali** [comment]
Thanks for reviewing.

> it's currently failing some of the existing tests.

Fixed that. I think I left before tests were done running last time and the forgot… Anyways afaict the remaining errors are deprecation errors on python 3.12, a download error, and pip refusing to install packages outside of virtual environments in the latest ubuntu.

> Related to that test - given that `openSUSE` returns `opensuse-tumbleweed` as the vendor, we probably want to transform that into `tumbleweed` so that the TOML parsing doesn't choke on a configuration section named `[tool.briefcase.app.myapp.linux.system.suse.opensuse-tumbleweed]`.

That seems to be parsed OK:
```
>>> toml.loads('''[tool.briefcase.app.myapp.linux.system.suse.opensuse-tumbleweed]
... key="val"
... ''')
{'tool': {'briefcase': {'app': {'myapp': {'linux': {'system': {'suse': {'opensuse-tumbleweed': {'key': 'val'}}}}}}}}}
```
I can still change it though there are a few other valid names (I think most notably opensuse-leap, sles)

> You'll also find that you'll at need some updates to the [briefcase-linux-system-template](https://github.com/beeware/briefcase-linux-system-template); as well as an update to [briefcase-template](https://github.com/beeware/briefcase-template) if you want openSUSE to work out of the box on new projects.

Will do.

> These are all relative minor changes, and I would have been able to make them on your behalf - but you've submitted the PR from your `main` branch rather than a feature branch, and as a result, it's not editable by maintainers.

The `Allow edit by maintainers` option is checked so it _should_ work

**Comment by rmartin16** [comment]
> > it's currently failing some of the existing tests.
> 
> Fixed that. I think I left before tests were done running last time and the forgot… Anyways afaict the remaining errors are deprecation errors on python 3.12, a download error, and pip refusing to install packages outside of virtual environments in the latest ubuntu, which are unrelated to this PR.

Externally managed environment strikes again...this time on Arch. I'll try to push a fix today.

```
Docker| #7 [4/7] RUN python3 -m ensurepip
Docker| #7 1.818 error: externally-managed-environment
Docker| #7 1.818 
Docker| #7 1.818 × This environment is externally managed
Docker| #7 1.818 ╰─> To install Python packages system-wide, try 'pacman -S
Docker| #7 1.818     python-xyz', where xyz is the package you are trying to
Docker| #7 1.818     install.
Docker| #7 1.818     
Docker| #7 1.818     If you wish to install a non-Arch-packaged Python package,
Docker| #7 1.818     create a virtual environment using 'python -m venv path/to/venv'.
Docker| #7 1.818     Then use path/to/venv/bin/python and path/to/venv/bin/pip.
Docker| #7 1.818     
Docker| #7 1.818     If you wish to install a non-Arch packaged Python application,
Docker| #7 1.818     it may be easiest to use 'pipx install xyz', which will manage a
Docker| #7 1.818     virtual environment for you. Make sure you have python-pipx
Docker| #7 1.818     installed via pacman.
Docker| #7 1.818 
Docker| #7 1.818 note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
Docker| #7 1.818 hint: See PEP 668 for the detailed specification.
```

File: changes/1416.feature.rst
Comment (freakboy3742) [inline changes/1416.feature.rst:1]:
```suggestion
Support for OpenSuSE Linux distributions was added.
```

File: docs/reference/platforms/linux/system.rst
Comment (freakboy3742) [inline docs/reference/platforms/linux/system.rst:135]:
```suggestion
* Vendor base (e.g., ``debian``, ``rhel``, ``arch``, ``suse``) 
* Vendor (e.g, ``debian``, ``ubuntu``, ``rhel``, ``fedora``, ``opensuse-tumbleweed``,
  ``arch``, ``manjaro``). The vendor identifier *may* be the same as the vendor base
  (e.g, in the case of Debian, Red Hat, or Arch)
```

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Looks good - I've provided a couple of tweaks to the documentation text, but otherwise this looks great. 

Since the `opensuse-tumbleweed` identifier works as is, I don't think any transformation is needed. Better to keep the original descriptor - it avoids us needing to document a special case, and when some distribution eventually uses `tumbleweed` as a version codename (and Murphy's law says its Gonna Happen One Day), we'll be safe from ambiguity.

The only pieces that are missing now are the template modifications; as is, this will dutifully *accept* a SUSE distribution... and then not build for it because the Dockerfile template is mostly empty.

**Review by freakboy3742 — APPROVED** [review]
I've made a couple more minor tweaks to documentation (including resolving some merge conflicts) - but otherwise this looks great! Thanks for the contribution!

**Pull request merged**
